### PR TITLE
build: migrate the CRDs to v1

### PR DESCRIPTION
**Pull request by SteveMattar**
Use `controller-gen` to generate the CRDs in v1.

Signed-off-by: Steve Mattar <smattar@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ PKG_MAN_OPTS ?= $(PKG_FROM_VERSION) $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
 # Image URL to use all building/pushing image targets
 IMG ?= quay.io/submariner/submariner-operator:$(VERSION)
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
```
Comment (skitt) [inline Makefile:57]: This needs to be changed too ;-).

**Review by skitt — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (SteveMattar) [reply Makefile:57]: the IMG ? I use it to replace the placeholder in the CSV file

Comment (skitt) [reply Makefile:57]: I’d picked line 57, not 56 — the “Produce CRDs that work back to Kubernetes 1.11 (no version conversion)” is wrong now.

Comment (SteveMattar) [reply Makefile:57]: wow good catch! +1

File: Makefile [hunk 2]
```diff
@@ -54,8 +54,8 @@ PKG_MAN_OPTS ?= $(PKG_FROM_VERSION) $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
 
 # Image URL to use all building/pushing image targets
 IMG ?= quay.io/submariner/submariner-operator:$(VERSION)
-# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true"
+# Produce CRDs v1 that doesn't work back to Kubernetes 1.11
```
Comment (dfarrell07) [inline Makefile:57]:
```suggestion
# Produce v1 CRDs that are compatible with Kubernetes >1.11
```

Is this what you're trying to say?

Comment (skitt) [inline Makefile:57]:
```suggestion
# Produce v1 CRDs, requiring Kubernetes 1.16 or later
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (SteveMattar) [reply Makefile:57]: better!

**Comment by dfarrell07** [comment]
I'm not sure what's going wrong, but these failures seem like real errors, not flaky failures or failures due to something else.

**Comment by SteveMattar** [comment]
@dfarrell07 yes I'm on it. Perhaps I should convert this PR to draft

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**